feature-utils/poly-look/src/react-components/filterChips.jsx

```diff
@@ -61,14 +61,6 @@ const FilterChips = ({
         )
   );
 
-  const [initialDefaultChips, setInitialDefaultChips] =
-    useState(defaultActiveChips);
-
-  if (defaultActiveChips !== initialDefaultChips) {
-    setActiveChips(defaultActiveChips);
-    setInitialDefaultChips(defaultActiveChips);
-  }
-
```

Comment:
Here's what I think this is for:

1. The user of this can pass in `defaultActiveChips` as a prop, which is supposed to active certain chips when the component is first loaded. That's handled above.
2. Now, if the user _changes_ the prop to something else, this code is presumably supposed to reflect that change: I.e. discarding the current user selection and selecting only the (new) default chips.

I can see how this code could cause infinite rerenders - that's not how you compare arrays in JS, it's always `false`.

Reply:
True

Reply:
As for how you _do_ compare arrays in JS, here's one way that does the right thing in most cases I think:

```
JSON.stringify(defaultActiveChips) !== JSON.stringify(initialDefaultChips)
```

We could alternatively write a function to properly deep equal arrays, or use a library that has that - it's not really in ES6, AFAIK.